### DISCUSSION TITLE
RT Structure

### DISCUSSION THREAD

**Discussion by cstrickland-pag** [Enet4/dicom-rs#512] [Q&A]
Hi,

Are there any examples of how to use this library with contour data in RT Structure Dicom files? Any help or suggestions on what to look into would be greatly appreciated.

Thanks,
Chris

**Comment by Enet4** [comment]
Thank you for your interest in DICOM-rs. The easiest part is loading the DICOM file as an object, of which there are many examples already. For RT structure files, I would say that consulting the standard for the modules that you are looking for from the IOD is key. In this case, the RT Structure Set IOD comprises the [ROI Contour module](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.8.8.6.html), and among the list of attributes we have _ROI Contour Sequence_, which can be consulted with something like this:

```rust
use dicom_object::open;
use dicom_dictionary::std::tags;

let file = open("file.dcm")?;

let contours = file.element(tags::ROI_CONTOUR_SEQUENCE)?;

for contour in contours.items().flatten() {
    let referenced_roi_number: i32 = contour.element(tags::REFERENCED_ROI_NUMBER)?.to_int()?;
    let source_pixel_planes = contour.get(tags::SOURCE_PIXEL_PLANES_CHARACTERISTICS_SEQUENCE)?;
    // ...
}
```

The documentation on [`DicomValue`](https://docs.rs/dicom-core/latest/dicom_core/value/enum.Value.html) might help you navigate the object once you retrieve a specific element through methods such as `get` or `element`. The only difficulty is in having to handle deeply nested data sets, but with this guidance it should hopefully be clearer what to do next.
